src/java/org/apache/cassandra/index/sai/disk/vector/CassandraDiskAnn.java

```diff
@@ -247,16 +253,16 @@ else if (compressedVectors == null)
                          ? VectorSimilarityFunction.COSINE
                          : similarityFunction;
                 var asf = compressedVectors.precomputedScoreFunctionFor(queryVector, sf);
-                var rr = view.rerankerFor(queryVector, similarityFunction);
+                var rr = !isRerankless ? view.rerankerFor(queryVector, similarityFunction) : null;
                 ssp = new SearchScoreProvider(asf, rr);
             }
             long start = System.nanoTime();
             var result = searcher.search(ssp, limit, rerankK, threshold, context.getAnnRerankFloor(), ordinalsMap.ignoringDeleted(acceptBits));
             long elapsed = System.nanoTime() - start;
             if (V3OnDiskFormat.ENABLE_RERANK_FLOOR)
                 context.updateAnnRerankFloor(result.getWorstApproximateScoreInTopK());
-            Tracing.trace("DiskANN search for {}/{} (usePruning: {}) visited {} nodes, reranked {} to return {} results from {}",
-                          limit, rerankK, usePruning, result.getVisitedCount(), result.getRerankedCount(), result.getNodes().length, source);
+            Tracing.trace("DiskANN search for {}/{} rerankless={}, usePruning: {} visited {} nodes, reranked {} to return {} results from {}",
```

Comment:
Nit: I'd use either
```suggestion
            Tracing.trace("DiskANN search for {}/{} rerankless: {}, usePruning: {} visited {} nodes, reranked {} to return {} results from {}",
```
or
```suggestion
            Tracing.trace("DiskANN search for {}/{} rerankless={}, usePruning={} visited {} nodes, reranked {} to return {} results from {}",
```